rubyforgood/diaper
# Validate and confirm item quantity for large donations

Resolves #1090 

### Description

This PR adds a confirmation message to verify unusually large donations.

This hardcodes the "large donation" limit (100,000). This also adds model validations for invalid values for 4-byte signed integers (in ruby, > 2e31 or < -2e31-1). I added those "magic numbers" as named constants in the `line_items` and `inventory_items`.

I haven't used CoffeeScript (and barely use front-end JS), so my JS validation might not be following best practices.

### Type of change

* Bug fix (add model validation)
* New feature  (add confirm dialog)
* Code style (delete unused asset files)

### How Has This Been Tested?

* New model tests (for validations)
* New feature test (for confirm dialog)
* Manual verification:
    - Log in
    - New donation
    - Fill out required form values
    - Verify functionality:
        - Add small item quantity -> *submits successfully*
        - Add unusually large item quantity -> *confirm dialog appears, submits successfully*
        - Add several items, including a large quantity ->  *confirm dialog appears, submits successfully*

### Screenshots

![Confirmation Screen](https://user-images.githubusercontent.com/814638/61976647-4f41c900-afba-11e9-90b4-884f3c764234.png)
